pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -121,6 +130,38 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 	return a.Client().Status().Patch(ctx, bastion, patch)
 }
 
+func getNetAndSubnet(ctx context.Context, a *actuator, cluster *extensions.Cluster) (vNet, subnet string, err error) {
+	var infrastructureStatus *gcp.InfrastructureStatus
+	worker := &extensionsv1alpha1.Worker{}
+
+	err = a.Client().Get(ctx, client.ObjectKey{Namespace: cluster.ObjectMeta.Name, Name: cluster.Shoot.Name}, worker)
```

Comment:
Why are you fetching the `worker` just to get the `InfrastructureStatus`?
Would it not be better to fetch the `Infrastructure` itself and check if it contain the `InfrastructureStatus` in `.status.providerStatus` section?

Reply:
I don't how to mock infra status information on my local cluster to verify it and mock integration test either. then I pick up `worker` and will take a look using `Infrastructure`

Reply:
had a private discussion with @dkistner there is some difficulty in the APJ region to download images from `eu.gcr.io` at the moment from home. It is extremely slow to download the terraform image when I develop/test the code in my local environment if I need to use the `Infrastructure` object. and integration test is the same situation.  also confirmed with @dkistner "A shoot can only have a machine (if joined or not) if the Worker resource exists."